pkg/splunk/client/azureblobclient.go

```diff
@@ -0,0 +1,238 @@
+	// are built.
+
+	// var err error
+
+	// Enforcing minimum version TLS1.2
+	// tr := &http.Transport{
+	// 	TLSClientConfig: &tls.Config{
+	// 		MinVersion: tls.VersionTLS12,
+	// 	},
+	// }
+	// tr.ForceAttemptHTTP2 = true
+	// httpClient := http.Client{Transport: tr}
+
+	// config := &Config{
+	// 	MaxRetries: 3,
+	// 	HTTPClient: httpClient,
+	// }
+
+	// azureBlobClient, err = NewAzureSession(ctx, appAzureBlobEndPoint, storageAccountName, secretAccessKey, config)
+	// if err != nil {
+	// 	scopedLog.Info("Error creating new AzureBlob Client Session", "err", err)
+	// 	return nil
+	// }
+
+	azureBlobClient = SplunkAzureBlobClientTemp{client: http.Client{}}
+	return &azureBlobClient
+
+}
+
+// GetAppsList get the list of apps from remote storage
+func (client *AzureBlobClient) GetAppsList(ctx context.Context) (RemoteDataListResponse, error) {
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("GetAppsList")
+
+	scopedLog.Info("Getting Apps list", " S3 Bucket", client.BucketName, "Prefix", client.Prefix)
+
+	azureBlobClient := client.Client
+
+	// Create a bucket list command for all files in bucket
+
+	//TODO: revise it in next sprint. had to use map temporary to let build work
+	opts := make(map[string]string)
+
+	opts["Prefix"] = client.Prefix
+	opts["MaxEnteries"] = "4000"
+
+	// List all objects from a bucket-name with a matching prefix.
+	listResponse, err := azureBlobClient.ListApps(context.Background(), client.BucketName, opts)
+
+	//TODO actual working logic to be covered in CSPL XXXX list azure apps
+	remoteDataListResponse, err := extractResponse(listResponse)
+
+	if err != nil {
+		scopedLog.Error(err, "Failed to get list apps response", "Azure Blob Bucket", client.BucketName)
+		return remoteDataListResponse, err
+	}
+
+	return remoteDataListResponse, err
+}
+
+func extractResponse(listBlobsXML []byte) (RemoteDataListResponse, error) {
+	remoteDataListResponse := RemoteDataListResponse{}
+
+	err := json.Unmarshal(listBlobsXML, &(remoteDataListResponse.Objects))
+	if err != nil {
+		return remoteDataListResponse, err
+	}
+
+	return remoteDataListResponse, nil
+}
+
+// DownloadApp downloads an app package from remote storage
+func (client *AzureBlobClient) DownloadApp(ctx context.Context, downloadRequest RemoteDataDownloadRequest) (bool, error) {
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("DownloadApp").WithValues("remoteFile", downloadRequest.RemoteFile,
+		downloadRequest.LocalFile, downloadRequest.Etag)
+
+	file, err := os.Create(downloadRequest.LocalFile)
+	if err != nil {
+		scopedLog.Error(err, "Unable to create local file")
+		return false, err
+	}
+	defer file.Close()
+
+	azureBlobClient := client.Client
+
+	//TODO : revise in next sprint . these are just place holder to more
+	// specific data structure so we can pass download options to rest calls
+	downloadOptions := make(map[string]string)
+
+	downloadOptions["maxChunkSize"] = "5gb"
+	downloadOptions["timeout"] = "5sec"
```

Comment:
Are we considering to make them configurable? similarly for listing optoins as well.

Reply:
I will refine the use of options in the next sprint. My current thinking is that the Azure rest APIS provides options to break a big blob and download data in smaller chunks so it is possible to download apps in chunks. S3 sdk also has similar features, but we have not used them so far. 
Timeout setting is for http request timeouts, and I may just use the default values to start with (same as we did leave them to default in Aws /Minio)